### PR TITLE
Add slideshow demo_module.

### PR DESCRIPTION
**Pull request by applmak**

### CODE REVIEW

**Comment by applmak** [comment]
I signed it!

File: demo_modules/slideshow/slideshow.js [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    // Return JSON that can be transmitted to the client and can instantiate
+    // the strategy there.
+    return {};
+  }
+}
+
+class ClientLoadStrategy {
+  loadContent(content) {
+    // Loads content on the client sent from the server. Returns a promise that
+    // is resolved when the content is finished loading.
```
Comment (bmt) [inline demo_modules/slideshow/slideshow.js]: You should be returning a resolved promise here.

File: demo_modules/slideshow/slideshow.js [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+  constructor(config) {
+    super();
+    this.config = config;
+    
+    // Drive client API v2.
+    this.driveClient = null;
+  }
+  init() {
+    // Get an authenticated API. When init's promise is resolved, we succeeded.
+    return googleapis.getAuthenticatedClient().then((client) => {
```
Comment (bmt) [inline demo_modules/slideshow/slideshow.js]: can this reject? Maybe you should log something in that case too?

Comment (applmak) [inline demo_modules/slideshow/slideshow.js]: Done.

File: demo_modules/slideshow/slideshow.js (hunk 1 above)
Comment (applmak) [inline demo_modules/slideshow/slideshow.js]: Done.

File: demo_modules/slideshow/slideshow.js [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    
+    // Content downloaded so far.
+    this.content = [];
+  }
+  willBeShownSoon(deadline) {
+    // Start the load strategy initing.
+    let loadingComplete = Promise.all([
+      this.displayStrategy.init(),
+      this.loadStrategy.init().then(() => {
+        let loadMoreContent = (opt_paginationToken) => {
```
Comment (bmt) [inline demo_modules/slideshow/slideshow.js]: This code is pretty hard to follow with the overloading of the loadMoreContent name. Could you name the local variable something else?

File: demo_modules/slideshow/slideshow.js [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    this.displayStrategy.tick(this.content, time, delta);
+  }
+}
+
+class ImageClient extends ClientModuleInterface {
+  willBeShownSoon(container, deadline) {
+    this.surface = new Surface(container, wallGeometry);
+    this.initedPromise = new Promise((startInit, reject) => {
+      network.once('init', (config) => {
+        this.loadStrategy = parseClientLoadStrategy(config.load);
```
Comment (bmt) [inline demo_modules/slideshow/slideshow.js]: Handle null strategies here

File: demo_modules/slideshow/slideshow.js [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+    container.appendChild(loadedContent);
+  }
+}
+
+class ImageServer extends ServerModuleInterface {
+  constructor(config) {
+    super();
+    
+    // The load strategy for this run of the module.
+    this.loadStrategy = parseServerLoadStrategy(config.load);
```
Comment (bmt) [inline demo_modules/slideshow/slideshow.js]: You're also not handling the null strategies here.  Maybe that's fine and instead you should have parse* throw with a useful error message instead of generating that error message everywhere where it is called.

Comment (applmak) [inline demo_modules/slideshow/slideshow.js]: Done.

File: demo_modules/slideshow/slideshow.js (hunk 3 above)
Comment (applmak) [inline demo_modules/slideshow/slideshow.js]: Renamed to 'fetch'

File: demo_modules/slideshow/slideshow.js (hunk 4 above)
Comment (applmak) [inline demo_modules/slideshow/slideshow.js]: Moved handling to the parse* functions.
